### PR TITLE
SNOW-1623450: fix flaky test by mocking datetime now

### PR DESCRIPTION
**Pull request by sfc-gh-aling**
1. Which Jira issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   

   Fixes SNOW-1623450

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
      - [ ] If this test skips Local Testing mode, I'm requesting review from @snowflakedb/local-testing
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency
   - [ ] If this is a new feature/behavior, I'm adding the Local Testing parity changes.

3. Please describe how your code solves the related issue.

Fix flaky test due to the inherently unstable current time execution.
in this PR I decoupled the test:
1. add local testing current_timestamp/date/time test to use mock session only
2. update merge clause to be independent of current_timestamp/date/time

### CODE REVIEW

File: tests/integ/scala/test_update_delete_merge_suite.py [hunk 1]
```diff
@@ -367,39 +368,40 @@ def test_merge_with_insert_clause_only(session, local_testing_mode):
...
-    if local_testing_mode:
-        assert res[1][1].date() == res[2][1].date() == now_datetime.date()
-        assert res[1][2] == res[2][2] == now_datetime.date()
-        assert res[1][3].minute == res[2][3].minute == now_datetime.time().minute
-        assert res[1][3].hour == res[2][3].hour == now_datetime.time().hour
-    else:
-        # in live connection time can differ because server timezone and test machine timezone differ
-        # so that we do not test exact time
-        assert len(res[1]) == len(res[2]) == 4
+    with patch("snowflake.snowpark.mock._functions.datetime") as dt:
```
Comment (sfc-gh-yzou) [inline tests/integ/scala/test_update_delete_merge_suite.py:371]: it is kind of wired that we are using mock for integration test? what is this part of code trying to test? does it has to use datetime.now()?

Comment (sfc-gh-yzou) [reply tests/integ/scala/test_update_delete_merge_suite.py:371]: or maybe you can move this to unit test if possible?

Comment (sfc-gh-aling) [reply tests/integ/scala/test_update_delete_merge_suite.py:371]: @sfc-gh-yzou  you have a valid point, this test is overloaded for multiple purpose -- testing merge + current timeout implementation in local testing and could be decoupled.

I will refactor it a bit

Comment (sfc-gh-aling) [reply tests/integ/scala/test_update_delete_merge_suite.py]: updated

**Review by sfc-gh-aalam — APPROVED** [review]

**Pull request merged**